environs/instances/instancetype.go

```diff
@@ -210,3 +211,29 @@ func (s byMemory) Less(i, j int) bool {
 	// Result is in descending order of cost so instance with lowest cost is used.
 	return inst0.Cost > inst1.Cost
 }
+
+//by Type is used to sort a slice by name by best effort. As we have different separators for different providers
+//we do it by best effort using "-" and "."
+//A possible sort could be:
+//a1.2xlarge, a1.large, lexical sort would put a1.2xlarge before a1.large -- aws
+
+type ByName []InstanceType
```

Comment:
Need some tests for this. Would be nice to get a few different provider samples.

Reply:
:+1:

Reply:
Added some test cases.
I did add changes to:
azure, aws, openstack and google.